### ISSUE TITLE
Add support for OAuth2

### ISSUE THREAD

**Issue by itwolf81** [jcustenborder/kafka-connect-email#5]
This version of IMAP Email Kafka Connector is perfect except it cannot connect to Office 365 IMAP exchange email account. My workaround was to implement my own Kafka connector based on your implementation using javax.mail dependency (version: 1.6.2) which works great with basic authentication. 

```
<dependency>
  <groupId>com.sun.mail</groupId>
  <artifactId>javax.mail</artifactId>
  <version>1.6.2</version>
</dependency>
```

Nevertheless, Basic authentication is being deprecated by Microsoft from 1.10.2022 and anytime could be removed. So correct way to connect to email should be using OAuth2 token.

**Comment by ghost** [comment]
Hi, is there being worked on OAuth2 integration?

**Comment by itwolf81** [comment]
> Hi, is there being worked on OAuth2 integration?

I don't think so. There were no updates ever since on this repo and even on Confluent Hub this connector isn't available anymore.
